### PR TITLE
feat: Implement barebones Server Subscriptions support.

### PR DESCRIPTION
**Pull request by FayeDel**
## About

This PR title essentially explains itself.

Server Subscriptions, formerly Premium Membership, have been released widespread towards all community-based servers that resides in the US, hence the ready-to-review PR.

This correlates to an earlier PR: #630 

## Checklist

- [X] I've ran `pre-commit` to format and lint the change(s) made.
- [X] I've checked to make sure the change(s) work on `3.8.6` and higher.
- [ ] This fixes/solves an [Issue](https://github.com/goverfl0w/discord-interactions/issues).
  - (If existent):
- [X] I've made this pull request for/as: (check all that apply)
  - [X] Documentation
  - [ ] Breaking change
  - [X] New feature/enhancement
  - [ ] Bugfix

### CODE REVIEW

**Comment by i0bs** [comment]
Premium memberships now offer information on multiple subscriptions present on a guild member. See [here for more.](https://support.discord.com/hc/en-us/articles/4415163187607-Premium-Memberships-for-Servers)

**Comment by FayeDel** [comment]
Marking this ready to review *only* because the feature's live;
(This PR still needs to be updated to latest unstable however)

File: interactions/api/models/role.py [hunk 1]
```diff
@@ -18,13 +18,27 @@ class RoleTags(DictSerializerMixin):
...
+    # TODO: Figure out what actual type it returns by Discord for some slots.
+
+    id: Optional[Snowflake] = field(converter=Snowflake, default=None)
     bot_id: Optional[Snowflake] = field(converter=Snowflake, default=None)
     integration_id: Optional[Snowflake] = field(converter=Snowflake, default=None)
     premium_subscriber: Optional[Any] = field(default=None)
-
-    # TODO: Figure out what actual type it returns, all it says is null.
+    subscription_listing_id: Optional[Snowflake] = field(converter=Snowflake, default=None)
+    purchasable_or_has_subscribers: Optional[Any] = field(default=None)
```
Comment (i0bs) [inline interactions/api/models/role.py]: Appears to return `bool`

File: interactions/api/models/role.py [hunk 2]
```diff
@@ -19,16 +19,19 @@ class RoleTags(DictSerializerMixin):
...
     A class object representing the tags of a role.
 
-    :ivar Optional[Snowflake] bot_id: The id of the bot this role belongs to
-    :ivar Optional[Snowflake] integration_id: The id of the integration this role belongs to
-    :ivar Optional[Any] premium_subscriber: Whether if this is the guild's premium subscriber role
+    :ivar Optional[Snowflake] bot_id?: The id of the bot this role belongs to
+    :ivar Optional[Snowflake] integration_id?: The id of the integration this role belongs to
+    :ivar Optional[Any] premium_subscriber?: Whether if this is the guild's server subscription role
+    :ivar Optional[Snowflake] subscription_listing_id?: The id of the linked server subscription role, if any.
+    :ivar Optional[bool] purchasable_or_has_subscribers?: Whether the role can be purchasable via server subscription or that the role contains subscribers, if any.
```
Comment (Damego) [inline interactions/api/models/role.py]: We refused from `?` in docs

File: interactions/api/models/gw.py [hunk 3]
```diff
@@ -258,10 +258,14 @@ class GuildJoinRequest(DictSerializerMixin):
 
     :ivar Snowflake user_id: The user ID of the event.
     :ivar Snowflake guild_id: The guild ID of the event.
+    :ivar Optional[Any] request: The actual request representing the event. This pertains primarily to _CREATE, but maybe _UPDATE as well.
+    :ivar Optional[str] status: The status of the event, which pertains to _CREATE.
     """
 
     user_id: Snowflake = field(converter=Snowflake)
     guild_id: Snowflake = field(converter=Snowflake)
+    request: Optional[Any] = field(default=None)
```
Comment (Damego) [inline interactions/api/models/gw.py:267]: There are no info about `request` to properly annotate it?

Comment (FayeDel) [reply interactions/api/models/gw.py:267]: The only thing I've gotten from my last testing (months ago) was a dictionary, however I don't have the logs for it nor is it documented in the docs (nor can I test it further at the moment) nor if I know if it's still there now compared to local testing.

Comment (Damego) [reply interactions/api/models/gw.py:267]: lol. Okay then

**Review by Damego — APPROVED** [review]
Looks good to me

**Pull request merged**